int32 num_links
rost_viewer_link_data[] link